lib/bap_disasm/disasm.cpp

```diff
@@ -124,7 +124,7 @@ class disassembler {
         };
     }
 
-    void set_memory(int64_t addr, const char *data, int offset, int length) {
+    void set_memory(int64_t addr, const uint8_t *data, int offset, int length) {
```

Comment:
the first thing that we should factor out is the changes made to the `lib/bap_disasm`. They should come as a separate PR.